### ISSUE TITLE
terminal command fails, crashing jupyter

### ISSUE THREAD

**Issue by mlshapiro** [mlshapiro/jupyterlab-flake8#7]
i've experienced some notebooks that get passed to the terminal, but the terminal never responds.

I added a "lint command timeout", which gets called attempting to cancel the lint command and dispose of the terminal.  This is reflected in the browser logs:

```bash
jupyterlab-flake8: getting notebook text index.js:532
jupyterlab-flake8: preparing lint command index.js:532
jupyterlab-flake8: sending lint command index.js:532
jupyterlab-flake8: lint command timed out index.js:532
jupyterlab-flake8: disposing flake8 and terminal
```

Unfortunately in some cases, jupyter locks up after this happens. I can't open any other files, and I have to restart the server. Interestingly, the server shutdown even fails. I get a "Shutdown confirmed" message, and then the it hangs up:

```bash
[I 11:23:42.939 LabApp] Saving file at Untitled.ipynb
^C[I 11:35:17.741 LabApp] interrupted
Serving notebooks from local directory: /...
1 active kernel
The Jupyter Notebook is running at:
http://localhost:8889/?token=9dba524af89e96f2e8c0541ca5f1501ae5b764a6728b1e3a
Shutdown this notebook server (y/[n])? y
[C 11:35:20.462 LabApp] Shutdown confirmed
```

This is all being caused by the command send to the terminal in line: 

```typescript
    this.log('sending lint command');
    this.term.session.send({type: 'stdin', content: [`${lint_cmd}\r`]})
```

The terminal is set up in function `load_linter()`:

```typescript
this.term = new Terminal({initialCommand: 'echo "Opening flake8 terminal"'});
try {
      this.term.session = await this.app.serviceManager.terminals.startNew();
...
```

and then the lint callback is configured in `activate_flake8()`:

```typescript
  activate_flake8() {
    // listen for stdout in onLintMessage
    this.term.session.messageReceived.connect(this.onLintMessage, this);
  }
```

The `onLineMessage` callback is never getting called in this circumstance, so my lint command is getting absorbed by the terminal and crashing jupyter.

**Comment by rgerkin** [comment]
I get the same errors.

**Comment by AlexanderPashuk** [comment]
Hey, guys.

I ran into similar problem and find out that in my case the issue was in validating the line:

```python
TGT_REGEX = 'action=\"https://test\.org/cas/v1/tickets/(?P<tgt>.*)\" method'
```

When I remove `"` from line above - Jupyter stops to freeze when I open the notebook.

**Comment by AlexanderPashuk** [comment]
**Reproduce**

Simple example to reproduce: create a notebook with content:

```python
print('\"')  # with escaped double quote
```

Then extension tries to send the following content:

```bash
(echo "print ('\\"')" | flake8 --exit-zero - && echo "@jupyterlab-flake8 finished linting" ) || (echo "@jupyterlab-flake8 finished linting failed")
```

But this bash command failed with:

![image](https://user-images.githubusercontent.com/3505985/53800667-476f8280-3f4e-11e9-9c0e-ebdeeb8a6210.png)

Correct bash command:

```bash
(echo "print ('\\\"')" | flake8 --exit-zero - && echo "@jupyterlab-flake8 finished linting" ) || (echo "@jupyterlab-flake8 finished linting failed")
```

**Possible solution**

Also escape `\` symbol.

Modify this line: https://github.com/mlshapiro/jupyterlab-flake8/blob/master/src/index.ts#L263

From:

```js
let escaped = contents.replace(/["`]/g,'\\$&');
```

To: 

```js
let escaped = contents.replace(/["`\\]/g,'\\$&');
```

**Comment by mlshapiro** [comment]
@AlexanderPashuk thank you very much for the suggestion. implemented in 0.3.1 - hopefully this helps. 

Lets close this for now and re-open if we still have problems.
